cdap-ui/app/cdap/components/Cloud/Profiles/CreateView/index.js

```diff
@@ -89,13 +96,34 @@ class ProfilesCreateView extends Component {
     document.querySelector('#header-namespace-dropdown').style.display = 'inline-block';
   }
 
-  onValueChange = (property, value) => {
+  onPropertyValueChange = (property, value) => {
     this.properties[property].value = value;
+    this.isCreateEnabled();
   };
 
   onMetadataChange = (metadata, e) => {
     this.setState({
       [metadata]: e.target.value
+    }, this.isCreateEnabled);
+  };
+
+  togglePropertyLock = (propertyName) => {
+    this.properties[propertyName].isEditable = !this.properties[propertyName].isEditable;
+  };
+
+  isCreateEnabled = () => {
+    let requiredPropertiesUnFullFilled = Object.keys(this.properties)
```

Comment:
nit: name this `unfilledRequiredProperties`? At first reading this i thought the variable was supposed to be a boolean.